### ISSUE TITLE
Incorporating the SVF into my own analysis pass

### ISSUE THREAD

**Issue by jjang3** [SVF-tools/SVF#881]
Dear developers,

First of all, thank you very much for providing such a great open-source tool so we can use it for our research.

I have been trying to incorporate the SVF for my analysis pass, and I am having difficulty calling the `SVF` pass for my analysis.

I have followed this example on how to use SVF APIs https://github.com/SVF-tools/SVF-example/blob/master/src/svf-ex.cpp. Furthermore, I can call all of those functions in the `svf-ex.cpp` into my own analysis code and **successfully** build the analysis library.

The problem is that when I try to use the `libSvf.a` library inside of my own analysis pass, I cannot successfully run the pass.

Let me try to visually explain what I mean above. For instance, here is the tree directory of my current LLVM analysis (`test_analysis`) directory:
```
./
├── test (contains my own .cpp / .h / CMake files for analysis)
├── SVF (git cloned SVF directory)
├── build
|      ├── test
|           └── libtest.so (this is the library of my own analysis pass which contains SVF functions)
|      └── SVF
|           └── build
|                 └── libSvf.a (this is the SVF library that I built along with my own analysis pass)
└── testing_input.ll (input .ll file which is used for opt command)
```

Typically, I just simply run the below command to run my `test` analysis:
`opt -load-pass-plugin /home/user/test_analysis/build/lib/libtest.so -passes="test" -S testing_input.ll -o testing_input_modified.ll`

However, after incorporating `SVF` related codes (for example, just simply adding even include statements below)
```
#include "SVF-FE/LLVMUtil.h"
#include "Graphs/SVFG.h"
#include "WPA/Andersen.h"
#include "SVF-FE/SVFIRBuilder.h"
#include "Util/Options.h"
```

I now get an error that says `opt: unknown pass name 'test'`; in other words, my own pass named `test`, for some reason, is now not able to be used as a pass.

So I am wondering if I were to use the `SVF` library (`libSvf.a`) for my analysis code, is there any additional `opt` command I need to add on top of my analysis command to make it work? or do I need to register the `SVF` pass inside of my own analysis code? 

Please let me know if the question doesn't make sense in any way, I will try to clarify it.
Kind regards.

**Comment by jjang3** [comment]
Just a quick update that I was able to resolve this issue, so I will close it. If anyone were having a similar issue, given you have set the analysis directory, as shown above, first, create a `CMakeLists.txt` file on the root directory and compile both analyses separately at first using the command (to create `libSvf.a` file in the `build/SVF/build/libSvf.a`:
```
# Include subdirectories.
add_subdirectory(SVF)
add_subdirectory(test)
```

Next, for `test` analysis's `CMakeLists.txt` file (this is a separate one from the root directory), include the line similar to:
```
target_link_libraries(test PUBLIC
  "${CMAKE_SOURCE_DIR}/build/SVF/lib/libSvf.a"
  "${test}"
)
```

In summary, it was a linking problem related to `CMake` file

**Comment by NeyagapulaSiddhartha** [comment]
@jjang3  could you please tell me how to use SVF in our analysis pass . Tipically i want something like  i have my own LLVM built from source and i have custom pass in llvm/lib/Transforms/   and i want to use flow sensitive multi threaded pointer analysis info in my custom pass
